hapi/src/services/eden-historic-delegates.service.js

```diff
@@ -0,0 +1,158 @@
+const {
+  edenDelegatesGql,
+  edenHistoricElectionGql,
+  edenElectionGql
+} = require('../gql')
+const { hasuraUtil, sleepUtil, eosUtil, axiosUtil } = require('../utils')
+const { hyperionConfig, edenConfig } = require('../config')
+
+const historicDelegates = async ({
+  next_key: nextKey = null,
+  limit = 100
+} = {}) => {
+  return await eosUtil.getTableRows({
+    code: edenConfig.edenContract,
+    scope: 0,
+    table: 'distaccount',
+    limit,
+    lower_bound: nextKey
+  })
+}
+
+const getActions = async (params, filter) => {
+  const limit = 100
+  const { data } = await axiosUtil.get(
+    `${hyperionConfig.api}/v2/history/get_actions`,
+    {
+      params: {
+        ...params,
+        limit,
+        filter: filter,
+        sort: 'asc',
+        simple: true,
+        checkLib: true
+      }
+    }
+  )
+  const notIrreversible = data.simple_actions.find(item => !item.irreversible)
+
+  if (notIrreversible) {
+    await sleepUtil(1)
+
+    return getActions(params)
+  }
+
+  return {
+    hasMore: data.total.value > limit + params.skip || 0,
+    actions: data.simple_actions
+  }
+}
+
+const registerHistoricDelegate = async (date, account, rank) => {
+  const election = await edenHistoricElectionGql.get({
+    date_election: { _lte: date }
+  })
+
+  const memberData = {
+    account: account
+  }
```

Comment:
Because if use only a variable string on edenDelegateGql.save, show this error "expected an object for type 'eden_delegates_insert_input', but found a string"}

Reply:
I changes it to { account } and work, thanks for observation